### PR TITLE
[ci] Add a helper script for Dockerfile generation.

### PR DESCRIPTION
**Pull request by rexwangcc**
Related issue = #

- Implement a helper script for easier Dockerfile generation.

### CODE REVIEW

File: ci/Dockerfile_generator.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        def f(os: str, version: str) -> str:
+            info(f"OS: {os}, version: {version}")
+            base_block = CPU_BASE_BLOCK.format(os=os, version=version)
+            scripts_block = SCRIPTS_BLOCK.format(
+                script=f"{os}_build_test_cpu.sh")
+            install_block = CPU_APT_INSTALL_BLOCK
+
+            # ubuntu 18.04 needs special treatments
+            if os == "ubuntu" and version == "18.04":
+                install_block += """ \\
```
Comment (qiao-bo) [inline ci/Dockerfile_generator.py:245]:
```suggestion
                install_block = install_block.rstrip() + """ \\
```

File: ci/Dockerfile_generator.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+        def f(os: str, version: str) -> str:
+            info(f"OS: {os}, version: {version}")
+            base_block = GPU_BASE_BLOCK.format(version=version)
+            scripts_block = SCRIPTS_BLOCK.format(script=f"{os}_build_test.sh")
+            install_block = GPU_APT_INSTALL_BLOCK
+
+            # ubuntu 20.04 needs special treatments
+            if os == "ubuntu" and version == "20.04":
+                install_block += """ \\
```
Comment (qiao-bo) [inline ci/Dockerfile_generator.py:267]:
```suggestion
                install_block = install_block.rstrip() + """ \\
```

**Review by qiao-bo — COMMENTED** [review]
Gerat, thanks! just some nits newline thingy

**Comment by qiao-bo** [comment]
@rexwangcc do we need to commit the changes on the Dockerfiles as well?

**Comment by rexwangcc** [comment]
> @rexwangcc do we need to commit the changes on the Dockerfiles as well?

Great question! I prefer to do it since the updated Dockerfiles are the results of the script, and we need to do it anyways from when we start to use the script, what do you think?

**Comment by qiao-bo** [comment]
> > @rexwangcc do we need to commit the changes on the Dockerfiles as well?
> 
> Great question! I prefer to do it since the updated Dockerfiles are the results of the script, and we need to do it anyways from when we start to use the script, what do you think?

sounds good, are they also the latest? 'cause we had a change in lines.

**Comment by rexwangcc** [comment]
> > > @rexwangcc do we need to commit the changes on the Dockerfiles as well?
> > 
> > 
> > Great question! I prefer to do it since the updated Dockerfiles are the results of the script, and we need to do it anyways from when we start to use the script, what do you think?
> 
> sounds good, are they also the latest? 'cause we had a change in lines.

will give it a try, one sec.

**Review by qiao-bo — APPROVED** [review]
Great! Thanks a lot!

**Pull request merged**